hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneFsDelete.java

```diff
@@ -139,10 +140,25 @@ protected void processPath(PathData item) throws IOException {
       if (moveToTrash(item) || !canBeSafelyDeleted(item)) {
         return;
       }
-      if (!item.fs.delete(path, deleteDirs)) {
+      BasicRootedOzoneFileSystem.InnerDeleteResult innerDeleteResult = null;
+      if (item.fs.getUri().getScheme().equals(OZONE_OFS_URI_SCHEME)) {
+        BasicRootedOzoneFileSystem ofs = (BasicRootedOzoneFileSystem) item.fs;
+        innerDeleteResult = ofs.deleteKeys(path, deleteDirs);
+      } else {
+        if (!item.fs.delete(path, deleteDirs)) {
+          throw new PathIOException(item.toString());
+        }
+      }
+      if (innerDeleteResult != null && !innerDeleteResult.isSuccess()) {
         throw new PathIOException(item.toString());
       }
-      out.println("Deleted " + item + (trailing ? OZONE_URI_DELIMITER : ""));
+      if (innerDeleteResult != null && innerDeleteResult.isPartiallyDeleted()) {
+        out.println(
+            "Path might not have been deleted completely: " + item + (trailing ?
```

Comment:
instead of returning partial flag, can we check if any item left inside the key by checking ozone and then have log ?

Reply:
To check if any item is left inside the key we would need to make another call which would result in total 2 calls whenever there would be a delete request which I guess would increase the time complexity!

Reply:
Recursive delete - performance may not be issue for OBS / Legacy as its anyway iterating all the keys with multiple calls. This will simplify the logic.
@aryangupta1998 @ashishkumar50